## Mini-Proyecto EDA

Descripción del perfil de Jorge Luis

Perfil del Cliente

Nombre y Apellidos: Jorge Luis Gómez

Edad: 34 años

Situación familiar: Padre soltero de un niño de 6 años

Prioridades de vivienda:

    Ubicación en una zona con acceso a centros educativos públicos de calidad para garantizar la mejor educación posible a su hijo.
    Proximidad a parques o espacios verdes donde pueda disfrutar de actividades al aire libre con su hijo.
    Acceso cercano al transporte público, dado que no posee vehículo propio.

Resumen del Perfil:

Nuestro colega busca un entorno que combine comodidad, accesibilidad y un ambiente familiar. Su interés principal es asegurar un equilibrio entre la proximidad a servicios esenciales, la calidad de vida y la participación en actividades comunitarias que favorezcan la integración y socialización en su nueva ciudad.

In [24]:
import pandas as pd
import numpy as np
import ipywidgets as widgets
import matplotlib.pyplot as plt
import plotly.graph_objects as go


In [4]:
archivo = ("emt.csv")
sep =(";")

df_emt = pd.read_csv(archivo, sep=sep)
df_emt

Id. Parada  Cod. Via / Cod. Vía Num. Portal  Suprimida  \
0          397.0                  NaN          59          0   
1          948.0                  NaN          62          0   
2          475.0              12180.0           3          0   
3          484.0              18390.0         108          0   
4          495.0              18390.0          85          0   
...          ...                  ...         ...        ...   
1121      2296.0              33480.0          SN          0   
1122      2050.0              81672.0         S/N          0   
1123      2078.0              30210.0   23 DAVANT          0   
1124      2303.0              31350.0  2 (DAVANT)          0   
1125      2350.0              81175.0         S/N          0   

                             Denominació / Denominación Línies / Líneas  \
0                       Felipe Valls - Lluís Cano (397)           62,N3   
1                               Tavernes Blanques (948)           16,N2   
2                            Brasil - Artur Piera (475)              73   
3                            Cid - Passatge Vidal (484)        70,93,N4   
4                                   Cid - Marconi (495)           93,98   
...                                                 ...             ...   
1121  Amado Granell Mesado (imparell) - Antonio Ferr...              14   
1122         Centre d'Investigació Príncep Felip (2050)           15,95   
1123                                Parc Central (2078)               8   
1124               Andreu Alfaro - Louis Braille (2303)       12,28,N10   
1125                José Roca Coll - Músic Gomis (2350)              73   

               Pròximes Arribades / Proximas Llegadas  \
0      http://www.emtvalencia.es/QR.php?sec=est&p=397   
1      http://www.emtvalencia.es/QR.php?sec=est&p=948   
2      http://www.emtvalencia.es/QR.php?sec=est&p=475   
3      http://www.emtvalencia.es/QR.php?sec=est&p=484   
4      http://www.emtvalencia.es/QR.php?sec=est&p=495   
...                                               ...   
1121  http://www.emtvalencia.es/QR.php?sec=est&p=2296   
1122  http://www.emtvalencia.es/QR.php?sec=est&p=2050   
1123  http://www.emtvalencia.es/QR.php?sec=est&p=2078   
1124  http://www.emtvalencia.es/QR.php?sec=est&p=2303   
1125  http://www.emtvalencia.es/QR.php?sec=est&p=2350   

                                              geo_shape  \
0     {"coordinates": [-0.4191237742755592, 39.50142...   
1     {"coordinates": [-0.3654941003557221, 39.50687...   
2     {"coordinates": [-0.3951866421883474, 39.47071...   
3     {"coordinates": [-0.4032098687744392, 39.46960...   
4     {"coordinates": [-0.4132110636422129, 39.46923...   
...                                                 ...   
1121  {"coordinates": [-0.3593398778553022, 39.45139...   
1122  {"coordinates": [-0.3462388297099983, 39.45189...   
1123  {"coordinates": [-0.37700013635997937, 39.4582...   
1124  {"coordinates": [-0.3949398474370588, 39.49751...   
1125  {"coordinates": [-0.39858953880204734, 39.4592...   

                                  geo_point_2d  
0      39.501427898376356, -0.4191237742755592  
1        39.5068784141963, -0.3654941003557221  
2       39.47071907570776, -0.3951866421883474  
3       39.46960970393574, -0.4032098687744392  
4      39.469233328598364, -0.4132110636422129  
...                                        ...  
1121   39.451398438149184, -0.3593398778553022  
1122   39.451893108982496, -0.3462388297099983  
1123  39.458250528158615, -0.37700013635997937  
1124    39.49751250712938, -0.3949398474370588  
1125   39.45929152242264, -0.39858953880204734  

[1126 rows x 9 columns]

En primer lugar analizaremos los centros educativos que ya era la prioridad de nuestro colega.

In [8]:
archivo = ("centros-educativos-en-valencia.csv")
sep =(";")

df_colegios= pd.read_csv(archivo, sep=sep)
df_colegios

Geo Point  \
0    39.50261499907377, -0.41316999999999976   
1      39.5302359990736, -0.3807020000000016   
2    39.45169899907408, -0.37516600000000083   
3     39.47401799907396, -0.3823219999999999   
4     39.47543999907394, -0.3713410000000003   
..                                       ...   
529  39.48093999907391, -0.35487599999999997   
530  39.45861799907405, -0.40538200000000024   
531   39.48667399907386, -0.3945380000000012   
532   39.43629299907419, -0.3964560000000006   
533   39.47612999907393, -0.3816880000000005   

                                             Geo Shape    codcen  \
0    {"coordinates": [-0.41317, 39.50261499907377],...  46002301   
1    {"coordinates": [-0.380702, 39.5302359990736],...  46009009   
2    {"coordinates": [-0.375166, 39.45169899907408]...  46009368   
3    {"coordinates": [-0.382322, 39.47401799907396]...  46009371   
4    {"coordinates": [-0.371341, 39.47543999907394]...  46009526   
..                                                 ...       ...   
529  {"coordinates": [-0.354876, 39.48093999907391]...  46036992   
530  {"coordinates": [-0.405382, 39.45861799907405]...  46037005   
531  {"coordinates": [-0.394538, 39.48667399907386]...  46037121   
532  {"coordinates": [-0.396456, 39.43629299907419]...  46037145   
533  {"coordinates": [-0.381688, 39.47612999907393]...  46037236   

                                                dlibre  \
0            CENTRE PRIVAT NATIVIDAD DE NUESTRA SEÑORA   
1                           CEIP MANUEL GONZÁLEZ MARTÍ   
2                          CENTRE PRIVAT MARTÍ SOROLLA   
3                          CENTRE PRIVAT ESCUELAS PÍAS   
4        CENTRE PRIVAT ESCOLANÍA NTRA.SRA.DESAMPARADOS   
..                                                 ...   
529  CENTRE ENSENYAMENTS ESPORTIUS C.F. DE JUDO Y D.A.   
530                    CENTRE PRIVAT FP MEDAC VALÈNCIA   
531  CENTRE PRIVAT FP DE LA PROFESIÓN DE PANADERÍA ...   
532                  ESCOLA PRIVADA DE MÚSICA SEDAJAZZ   
533                     CENTRE ESTRANGER LABA VALENCIA   

                                            dgenerica_  \
0    CENTRE PRIVAT D'EDUCACIÓ INFANTIL, PRIMÀRIA I ...   
1              COL·LEGI D'EDUCACIÓ INFANTIL I PRIMÀRIA   
2         CENTRE PRIVAT D'EDUCACIÓ INFANTIL I PRIMÀRIA   
3    CENTRE PRIVAT D'EDUCACIÓ INFANTIL, PRIMÀRIA I ...   
4       CENTRE PRIVAT D'EDUCACIÓ PRIMÀRIA I SECUNDÀRIA   
..                                                 ...   
529         CENTRE AUTORITZAT D'ENSENYAMENTS ESPORTIUS   
530     CENTRE PRIVAT FORMACIÓ PROFESSIONAL ESPECÍFICA   
531     CENTRE PRIVAT FORMACIÓ PROFESSIONAL ESPECÍFICA   
532                           ESCOLA PRIVADA DE MÚSICA   
533                     CENTRE DOCENT PRIVAT ESTRANGER   

                                            despecific     regimen  \
0                          NATIVIDAD DE NUESTRA SEÑORA  CONCERTADO   
1                                MANUEL GONZÁLEZ MARTÍ     PÚBLICO   
2                                        MARTÍ SOROLLA  CONCERTADO   
3                                        ESCUELAS PÍAS  CONCERTADO   
4                      ESCOLANÍA NTRA.SRA.DESAMPARADOS  CONCERTADO   
..                                                 ...         ...   
529  CENTRO DE FORMACIÓN DE JUDO Y DISCIPLINAS ASOC...     PRIVADO   
530                                     MEDAC VALÈNCIA     PRIVADO   
531          DE LA PROFESIÓN DE PANADERÍA Y PASTELERÍA     PRIVADO   
532                                           SEDAJAZZ     PRIVADO   
533                                      LABA VALENCIA     PRIVADO   

                                  direccion  codpos municipio_ provincia_  \
0                          CL. ESPARTERO, 2   46100  Burjassot   València   
1                   CL. ALQUERIA GILET, S/N   46016   València   València   
2        CL. INGENIERO JOAQUÍN BENLLOCH, 45   46026   València   València   
3                         CL. CARNICEROS, 6   46001   València   València   
4                  CL

In [10]:
df_colegios_valencia = df_colegios.loc[(df_colegios.loc[:,"regimen"] == 'PÚBLICO') & (df_colegios.loc[:,"municipio_"] == 'València')]
df_colegios_valencia

Geo Point  \
1      39.5302359990736, -0.3807020000000016   
22   39.48130099907391, -0.32878499999999977   
23   39.45903399907406, -0.34142399999999995   
28   39.46112999907402, -0.36987699999999946   
29     39.467280999074, -0.39990700000000096   
..                                       ...   
478    39.47526399907393, -0.349202000000001   
480   39.46387699907402, -0.4018119999999993   
499   39.45339299907408, -0.3657030000000005   
502  39.467673999073995, -0.3999069999999993   
503   39.45254899907409, -0.3588799999999998   

                                             Geo Shape    codcen  \
1    {"coordinates": [-0.380702, 39.5302359990736],...  46009009   
22   {"coordinates": [-0.328785, 39.48130099907391]...  46011533   
23   {"coordinates": [-0.341424, 39.45903399907406]...  46011570   
28   {"coordinates": [-0.369877, 39.46112999907402]...  46012197   
29   {"coordinates": [-0.399907, 39.467280999074], ...  46012321   
..                                                 ...       ...   
478  {"coordinates": [-0.349202, 39.47526399907393]...  46028582   
480  {"coordinates": [-0.401812, 39.46387699907402]...  46029525   
499  {"coordinates": [-0.365703, 39.45339299907408]...  46033279   
502  {"coordinates": [-0.399907, 39.467673999073995...  46033334   
503  {"coordinates": [-0.35888, 39.45254899907409],...  46033437   

                                                dlibre  \
1                           CEIP MANUEL GONZÁLEZ MARTÍ   
22                                     CEIP MALVA-ROSA   
23                                        CEIP EL GRAU   
28                                   CEIP JAIME BALMES   
29                                   CEIP SECTOR AÉREO   
..                                                 ...   
478                   EI 1er CICLE MUNICIPAL PARDALETS   
480                        EI 1er CICLE MUNICIPAL SOLC   
499             EI 1er CICLE MUNICIPAL QUATRE CARRERES   
502                   EI 1er CICLE MUNICIPAL MINI-POLI   
503  EI 1er CICLE MUNICIPAL MESTRA EMPAR NAVARRO I ...   

                                  dgenerica_  \
1    COL·LEGI D'EDUCACIÓ INFANTIL I PRIMÀRIA   
22   COL·LEGI D'EDUCACIÓ INFANTIL I PRIMÀRIA   
23   COL·LEGI D'EDUCACIÓ INFANTIL I PRIMÀRIA   
28   COL·LEGI D'EDUCACIÓ INFANTIL I PRIMÀRIA   
29   COL·LEGI D'EDUCACIÓ INFANTIL I PRIMÀRIA   
..                                       ...   
478          ESCOLA INFANTIL DE PRIMER CICLE   
480          ESCOLA INFANTIL DE PRIMER CICLE   
499          ESCOLA INFANTIL DE PRIMER CICLE   
502          ESCOLA INFANTIL DE PRIMER CICLE   
503          ESCOLA INFANTIL DE PRIMER CICLE   

                                 despecific  regimen  \
1                     MANUEL GONZÁLEZ MARTÍ  PÚBLICO   
22                               MALVA-ROSA  PÚBLICO   
23                                  EL GRAU  PÚBLICO   
28                             JAIME BALMES  PÚBLICO   
29                             SECTOR AÉREO  PÚBLICO   
..                                      ...      ...   
478                     MUNICIPAL PARDALETS  PÚBLICO   
480                          MUNICIPAL SOLC  PÚBLICO   
499               MUNICIPAL QUATRE CARRERES  PÚBLICO   
502                     MUNICIPAL MINI-POLI  PÚBLICO   
503  MUNICIPAL MESTRA EMPAR NAVARRO I GINER  PÚBLICO   

                                  direccion  codpos municipio_ provincia_  \
1                   CL. ALQUERIA GILET, S/N   46016   València   València   
22                   CL. PICO DE TEIDE, 1-3   46011   València   València   
23                         AV. BALEARES, 64   46023   València   València   
28                  CL. MAESTRO AGUILAR, 15   46006   València   València   
29            CL. SANTA CRUZ DE TENERIFE, 1   46018   València   València   
..                                      ...     ...        ...        ...   
478                   CL. POETA ARTOLA, S/N   46021   València   València   
480                    CL. TRES FORQUES, 94   46018   València   València   
4

Selecciona solo aquellos colegios cuyo régimen es "PÚBLICO" y filtramos los colegios que están ubicados en el municipio de "València". Filtramos aquellos colegios cuyo nombre (almacenado en la columna dlibre) empieza con "CEIP" (lo que puede indicar que son centros de educación infantil y primaria, comúnmente en España).

In [15]:
df_colegios = df_colegios.loc[(df_colegios.loc[:,"regimen"] == 'PÚBLICO') & (df_colegios.loc[:,"municipio_"] == 'València') & (df_colegios["dlibre"].str.startswith("CEIP"))]
df_colegios_groupby = df_colegios.groupby('codpos').size().reset_index(name='num_colegios').sort_values(by = 'num_colegios',ascending = False)


Los resultados están ordenados de tal manera que los códigos postales con más colegios están al principio. Finalmente, al ejecutar df_colegios_groupby, se muestra el DataFrame con el número de colegios por código postal en el municipio de Valencia, filtrados por el régimen "PÚBLICO" y que empiezan con "CEIP".

In [16]:
df_colegios_groupby

codpos  num_colegios
18   46022             9
13   46017             7
10   46014             6
14   46018             6
15   46019             5
7    46011             5
11   46015             4
2    46006             4
17   46021             4
20   46024             3
19   46023             3
5    46009             3
9    46013             3
22   46026             3
16   46020             2
21   46025             2
23   46035             2
8    46012             2
6    46010             2
4    46008             2
3    46007             2
0    46003             1
12   46016             1
1    46004             1
24   46112             1

Escogemos las zonas donde hay más colegios y las buscamos mediante su código postal: 46022, 46017 y 46018

Esta zona es la 46022 y pertenece a los barrios de Beterós y La Carrasca con 9 centros educativos públicos

![zona_46022](46022.jpeg)

Esta zona es la 46017 y pertenece a los barrios de San Marcelino y La Torre, con 7 centros educativos públicos

![zona_46017](46017.jpeg)

Esta zona es la 46018 y pertenece a los barrios de Tres Forques y La FuenSanta con 6 centros educativos públicos

![zona_46018](46018.png)

Una vez hemos obtenido las 3 zonas con mayor cantidad de colegios públicos, nos pasamos a analizar qué zonas tienen una mayor cantidad de parques infantiles para que el pequeño se entretenga 

In [17]:
archivo=("zones-jocs-infantils-zona-juegos-infantiles.csv")
sep = (";")

df_parques= pd.read_csv(archivo, sep=sep)
df_parques

objectid    Codigo                        Jardin               Barrio  \
0          92  99101122            Iglesia de Vera Pl        La Malva-Rosa   
1          93  99101123          Carola Reig Salva Pl               Betero   
2          95  99101128  Serreria C/ - Justo Vilar C/  Cabanyal-Canyamelar   
3          96  99101130                    Remonta C/  Cabanyal-Canyamelar   
4         100  99101138                  JJ Sister C/              El Grau   
..        ...       ...                           ...                  ...   
488       485  99122006      Jardin del Turia Tramo 6      Cauce del Turia   
489       489  99122012     Jardin del Turia Tramo 12      Cauce del Turia   
490       490  99122012     Jardin del Turia Tramo 12      Cauce del Turia   
491        94  99101127             Marti Grajales C/  Cabanyal-Canyamelar   
492       102  99101141        Jesus de Medinaceli Pl              El Grau   

       DM   Zona         Mantenimie               Tipo  CREATED_USER  \
0    11.0  NORTE  SJ JARDINES NORTE  Juegos Infantiles           NaN   
1    11.0  NORTE  SJ JARDINES NORTE  Juegos Infantiles           NaN   
2    11.0  NORTE  SJ JARDINES NORTE  Juegos Infantiles           NaN   
3    11.0  NORTE  SJ JARDINES NORTE  Juegos Infantiles           NaN   
4    11.0  NORTE  SJ JARDINES NORTE  Juegos Infantiles           NaN   
..    ...    ...                ...                ...           ...   
488  20.0  TURIA               OAMJ  Juegos Infantiles           NaN   
489  20.0  TURIA               OAMJ  Juegos Infantiles           NaN   
490  20.0  TURIA               OAMJ  Juegos Infantiles           NaN   
491  11.0  NORTE  SJ JARDINES NORTE  Juegos Infantiles           NaN   
492  11.0  NORTE  SJ JARDINES NORTE  Juegos Infantiles           NaN   

     CREATED_DATE  LAST_EDITED_USER  LAST_EDITED_DATE  \
0             NaN               NaN               NaN   
1             NaN               NaN               NaN   
2             NaN               NaN               NaN   
3             NaN               NaN               NaN   
4             NaN               NaN               NaN   
..            ...               ...               ...   
488           NaN               NaN               NaN   
489           NaN               NaN               NaN   
490           NaN               NaN               NaN   
491           NaN               NaN               NaN   
492           NaN               NaN               NaN   

                                             geo_shape  \
0    {"coordinates": [-0.32970728183685255, 39.4842...   
1    {"coordinates": [-0.33857397813375084, 39.4704...   
2    {"coordinates": [-0.33447314272886003, 39.4670...   
3    {"coordinates": [-0.33072077575470954, 39.4746...   
4    {"coordinates": [-0.3338307525882248, 39.45926...   
..                                                 ...   
488  {"coordinates": [-0.3794848289408849, 39.48215...   
489  {"coordinates": [-0.35964027618013744, 39.4625...   
490  {"coordinates": [-0.3588234662409007, 39.46066...   
491  {"coordinates": [-0.33341094653734205, 39.4684...   
492  {"coordinates": [-0.33778834774135397, 39.4592...   

                                geo_point_2d  
0    39.48420206376716, -0.32970728183685255  
1    39.47041677471436, -0.33857397813375084  
2    39.46706223918951, -0.33447314272886003  
3    39.47464671747163, -0.33072077575470954  
4     39.45926791454811, -0.3338307525882248  
..                                       ...  
488  39.482153265204744, -0.3794848289408849  
489  39.46253577088889, -0.35964027618013744  
490   39.46066610288398, -0.3588234662409007  
491  39.46845661398144, -0.33341094653734205  
492  39.45928171916552, -0.33778834774135397  

[493 rows x 14 columns]

In [35]:
df_zona1 = df_parques.loc[(df_parques.loc[:, "Barrio"] == 'Betero') |  (df_parques.loc[:, "Barrio"] == 'La Carrasca')]
df_zona2 = df_parques.loc[(df_parques.loc[:, "Barrio"] == 'La Torre') |  (df_parques.loc[:, "Barrio"] == 'Sant Marceli')]
df_zona3 = df_parques.loc[(df_parques.loc[:, "Barrio"] == 'Tres Forques') | (df_parques.loc[:, "Barrio"] == 'La Fuensanta')]

In [37]:
df_zona1.loc[:,"Barrio"] = 'Betero y La Carrasca'
df_zona2.loc[:, "Barrio"] = 'La Torre y Sant Marceli'
df_zona3.loc[:,"Barrio"] = 'Tres Forques y La Fuensanta'

Para visualizar los datos, utilizaremos la librería Plotly

In [40]:
fig = go.Figure()

# Histograma para df_zona1
fig.add_trace(go.Histogram(x=df_zona1["Barrio"], name="Zona 1: Betero y La Carrasca", opacity=0.75))

# Histograma para df_zona2
fig.add_trace(go.Histogram(x=df_zona2["Barrio"], name="Zona 2: La Torre y Sant Marcel", opacity=0.75))

# Histograma para df_zona3
fig.add_trace(go.Histogram(x=df_zona3["Barrio"], name="Zona 3: Tres Forques y La Fuensanta", opacity=0.75))

# Configuración del diseño
fig.update_layout(
    title="Parques por Zona",
    xaxis_title="Área",
    yaxis_title="Cantidad de parques",
    barmode='overlay',  # Superponer los histogramas
    bargap=0.1,  # Espacio entre barras
    template="plotly_dark"  # Estilo del gráfico (opcional)
)
fig.show()

Una vez hemos obtenido los datos, nos quedamos con las dos zonas que tienen más parques: Zona 1 (Bétero y La Carrasca) y la Zona 2 (La Torre y Sant Marceli)

Por último analizaremos cuál de las dos zonas está mejor comunicada en cuanto a lineas de metro se refiere. Para ello, hemos obtenido los datos del Ayuntamiento de Valencia.

In [42]:
archivo=("fgv-bocas.csv")
sep = (";")

df_metro = pd.read_csv(archivo, sep=sep)
df_metro

gid  Denominació / Denominación Línies / Líneas  Id. Parada  Id. Boca  \
0    182                       FOIOS               3           6       215   
1    187                 RAFELBUNYOL               3           1       220   
2     22                   Á. GIMERÁ         1,3,5,9          17         3   
3     11                      ARAGÓN             5,7          24        13   
4      6                     AMISTAT             5,7          23        11   
..   ...                         ...             ...         ...       ...   
182  169          RIBA-ROJA DE TÚRIA               9         186       202   
183    2                     ALAMEDA         3,5,7,9          14         6   
184  120  MARINA REIAL JOAN CARLES I               5         126       153   
185  127              SEMINARI - CEU               1          78       160   
186  151                  CANTERER�A               1          56       184   

                Pròximes Arribades / Próximas llegadas  \
0    https://geoportal.valencia.es/geoportal-servic...   
1    https://geoportal.valencia.es/geoportal-servic...   
2    https://geoportal.valencia.es/geoportal-servic...   
3    https://geoportal.valencia.es/geoportal-servic...   
4    https://geoportal.valencia.es/geoportal-servic...   
..                                                 ...   
182  https://geoportal.valencia.es/geoportal-servic...   
183  https://geoportal.valencia.es/geoportal-servic...   
184  https://geoportal.valencia.es/geoportal-servic...   
185  https://geoportal.valencia.es/geoportal-servic...   
186  https://geoportal.valencia.es/geoportal-servic...   

                                     proximes_llegadas  \
0    https://geoportal.valencia.es/geoportal-servic...   
1    https://geoportal.valencia.es/geoportal-servic...   
2    https://geoportal.valencia.es/geoportal-servic...   
3    https://geoportal.valencia.es/geoportal-servic...   
4    https://geoportal.valencia.es/geoportal-servic...   
..                                                 ...   
182  https://geoportal.valencia.es/geoportal-servic...   
183  https://geoportal.valencia.es/geoportal-servic...   
184  https://geoportal.valencia.es/geoportal-servic...   
185  https://geoportal.valencia.es/geoportal-servic...   
186  https://geoportal.valencia.es/geoportal-servic...   

                                             geo_shape  \
0    {"coordinates": [-0.353869438543834, 39.537223...   
1    {"coordinates": [-0.33105832383704287, 39.5885...   
2    {"coordinates": [-0.38410946248914607, 39.4695...   
3    {"coordinates": [-0.3577954459515307, 39.47269...   
4    {"coordinates": [-0.3518176379178333, 39.47074...   
..                                                 ...   
182  {"coordinates": [-0.5594444278345464, 39.54333...   
183  {"coordinates": [-0.36549713547643425, 39.4736...   
184  {"coordinates": [-0.3258508441839875, 39.46325...   
185  {"coordinates": [-0.38974443064379294, 39.5499...   
186  {"coordinates": [-0.41207221163652963, 39.5023...   

                                 geo_point_2d  
0        39.5372238152284, -0.353869438543834  
1    39.588523863986794, -0.33105832383704287  
2     39.46959137449779, -0.38410946248914607  
3      39.47269126027633, -0.3577954459515307  
4      39.47074908629906, -0.3518176379178333  
..                                        ...  
182    39.54333495963937, -0.5594444278345464  
183   39.47363752703317, -0.36549713547643425  
184    39.46325302065733, -0.3258508441839875  
185   39.54998779211858, -0.38974443064379294  
186   39.50238037049734, -0.41207221163652963  

[187 rows x 9 columns]

Para realizar la búsqueda del transporte por zona, hemos buscado en google maps las lineas de metro que pasan por cada una de ellas, de esta mamenra sabremos qué datos filtrar en el df

En la zona 1, podemos observar que pasa tanto la línea de metro, como de tranvía. 

![zona_1](zona1.jpeg)

In [43]:
df_metro_zona1 = df_metro.loc[(df_metro.loc[:, "Línies / Líneas"] == '5,7') | (df_metro.loc[:, "Línies / Líneas"] == '4,6')].groupby('Denominació / Denominación')['Línies / Líneas'].value_counts().reset_index()
df_metro_zona1.iloc[:,0:2]

Denominació / Denominación Línies / Líneas
0                     AMISTAT             5,7
1                      ARAGÓN             5,7
2                       AYORA             5,7
3                   DR. LLUCH             4,6
4               EUGENIA VIÑES             4,6
5                   LA CADENA             4,6
6                 LA CARRASCA             4,6
7                   LA MARINA             4,6
8                  LES ARENES             4,6
9                PRIMADO REIG             4,6
10                   SERRERÍA             4,6
11                  TARONGERS             4,6
12    UNIVERSITAT POLITÈCNICA             4,6
13                V. ZARAGOZÁ             4,6

In [47]:
df_lineas_zona1 = df_metro_zona1.loc[(df_metro_zona1.loc[:, "Denominació / Denominación"] == "AMISTAT")| (df_metro_zona1.loc[:, "Denominació / Denominación"] == "TARONGERS") | (df_metro_zona1.loc[:, "Denominació / Denominación"] == "AYORA")| (df_metro_zona1.loc[:, "Denominació / Denominación"] == "LA CARRASCA")]

df_lineas_zona1

Denominació / Denominación Línies / Líneas  count
0                     AMISTAT             5,7      2
2                       AYORA             5,7      2
6                 LA CARRASCA             4,6      1
11                  TARONGERS             4,6      1

En la zona 2, podemos observar que sólo pasa la línea de metro 

![zona_2](zona2.jpeg)

In [50]:
df_metro_zona2 = df_metro.loc[(df_metro.loc[:, "Línies / Líneas"] == '1,3,7')].groupby('Denominació / Denominación')['Línies / Líneas'].value_counts().reset_index()
df_metro_zona2.iloc[:,0:2]

Denominació / Denominación Línies / Líneas
0                   HOSPITAL           1,3,7
1            JOAQUÍN SOROLLA           1,3,7
2                   PAIPORTA           1,3,7
3                    PATRAIX           1,3,7
4                    PICANYA           1,3,7
5                SANT ISIDRE           1,3,7
6                    TORRENT           1,3,7
7           TORRENT AVINGUDA           1,3,7
8               VALENCIA SUD           1,3,7

In [51]:
df_lineas_zona2 = df_metro_zona2.loc[(df_metro_zona2.loc[:, "Denominació / Denominación"] == "SANT ISIDRE")| (df_metro_zona2.loc[:, "Denominació / Denominación"] == "PATRAIX")]

df_lineas_zona2

Denominació / Denominación Línies / Líneas  count
3                    PATRAIX           1,3,7      2
5                SANT ISIDRE           1,3,7      1

In [59]:
df_lineas_zona1.loc[:,"Línies / Líneas"] = "Zona 1"
df_lineas_zona2.loc[:, "Línies / Líneas"] = "Zona 2"


In [63]:
df_lineas_zona1

Denominació / Denominación Línies / Líneas  count
0                     AMISTAT          Zona 1      2
2                       AYORA          Zona 1      2
6                 LA CARRASCA          Zona 1      1
11                  TARONGERS          Zona 1      1

In [64]:
fig = go.Figure()

# Histograma para df_zona1
fig.add_trace(go.Histogram(x=df_lineas_zona1["Línies / Líneas"], name= "Zona 1. Número de estaciones de metro.", opacity=0.75))

# Histograma para df_zona2
fig.add_trace(go.Histogram(x=df_lineas_zona2["Línies / Líneas"], name= "Zona 2. Número de estaciones de metro.", opacity=0.75))


# Configuración del diseño
fig.update_layout(
    title="Líneas de transporte por zona",
    xaxis_title="Área",
    yaxis_title="Cantidad de parques",
    barmode='overlay',  # Superponer los histogramas
    bargap=0.1,  # Espacio entre barras
    template="plotly_dark"  # Estilo del gráfico (opcional)
)
fig.show()

Finalmente, según los datos obtenidos hemos concluído que la mejor zona para que nuestro colega viva en base a sus necesidades en la zona 1, Betero y La Carrasca, ya que es la zona donde hay más centros educativas, más parques infantiles y además es la que mejor comunicada está ya que por ella pasan 4 líneas de metro y tranvía. Sin contar las numerosas líneas de autobuses las cuales no nos ha dado tiempo a analizar :)